src/pages/Payment/Payment.tsx

```diff
@@ -9,27 +9,87 @@ import PriceSummary from '../../components/organisms/PriceSummary/PriceSummary';
 import OptionWrapper from '../../components/molecules/OptionWrapper/OptionWrapper';
 import { FormattedMessage, useIntl } from 'react-intl';
 import { toast } from 'react-toastify';
+import { RootState } from '../../store';
+import { useDispatch, useSelector } from 'react-redux';
+import Card from '../../components/organisms/Card/Card';
+import { ChangeEvent } from 'react';
+import { saveCardToUser, saveOrderToUser } from '../../redux/user/UserActions';
+import { OrderItem } from '../../components/organisms/Order/Order';
+import { resetCart } from '../../redux/cart/CartAction';
 
 
 
 const Payment = () :JSX.Element => {
 
     const history = useHistory();
 
-    const [paymentOption, setPaymentOption] = useState('');
-    const [amount, setAmount] = useState('100');
+    const userState = useSelector<RootState, RootState["userState"]>((state: RootState) => state.userState);
+    const cartState = useSelector<RootState, RootState["cartState"]>((state: RootState) => state.cartState);
+    const calculateDiscountedPrice = (price: number, discountPercent: number) => (price - ((price * discountPercent) / 100));
+   
+    const {formatMessage} = useIntl();
+    const dispatch = useDispatch();
+    
+    const {user} = userState;
+    const {cartItems} = cartState;
+
+    let amount = 0;
+
+    const orderMappedCartItems: OrderItem[] = cartItems.map((item, index) => {
+        const {id, name, price, discountPercent, size, color, quantity} = item;
+        const discountedPrice =  calculateDiscountedPrice(price, discountPercent);
+        amount += discountedPrice * quantity;
+
+        return { 
+          productName: name,
+          productId: id,
+          size,
+          color, 
+          quantity}
+      });
+
+    const [paymentOption, setPaymentOption] = useState<string>('');
     const [showConfirmModal, setShowConfirmModal] = useState<boolean>(false);
-    const [refundId, setRefundId] = useState('');
+    const [refundId, setRefundId] = useState<string>('');
+
+    //For the card form
+
+    const [cardNumber, setCardNumber] = useState<string>('')
```

Comment:
Should this be a string or a number type?

Reply:
A 16 digit credit card number never has a number type :P If you want you can add validation for the same. Most welcome to work on it